streaming/consumer.cc

```diff
@@ -65,11 +68,13 @@ reader_consumer_v2 make_streaming_consumer(sstring origin,
                         cf->enable_off_strategy_trigger();
                     }
                     return cf->add_sstable_and_update_cache(sst, offstrategy);
-                }).then([cf, s, sst, use_view_update_path, &vb]() mutable -> future<> {
-                    if (!use_view_update_path) {
-                        return make_ready_future<>();
+                }).then([cf, s, sst, use_view_update_path, &vb, &vbw]() mutable -> future<> {
+                    if (use_view_update_path == db::view::sstable_destination_decision::staging_managed_by_vbc) {
+                        return vbw.local().notify();
+                    } else if (use_view_update_path == db::view::sstable_destination_decision::staging_directly_to_generator) {
+                        return vb.local().register_staging_sstable(sst, std::move(cf));
```

Comment:
see load_sstable_for_tablet() in streaming/stream_blob.cc. It's not registering the staging sstable when db::view::sstable_destination_decision::staging_directly_to_generator, nor it's notifying the view building worker. I assume you must patch it too, so the coordinator will be correct with file based streaming in tablet migration that bypasses this streaming consumer path.